### PR TITLE
fix(integrations): Braze adding dynamic support for eu data center

### PR DESCRIPTION
**Pull request by utsabc**
## Description of the change

Adding dynamic support for braze EU data center

## Type of change
- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests
- [x] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: v0/destinations/braze/transform.js [hunk 1]
```diff
@@ -371,10 +371,11 @@ function process(event) {
   // Init -- mostly for test cases
   destination.Config.endPoint = "https://rest.fra-01.braze.eu";
 
+  // Ref: https://www.braze.com/docs/user_guide/administrative/access_braze/braze_instances
   if (destination.Config.dataCenter) {
     const dataCenterArr = destination.Config.dataCenter.trim().split("-");
     if (dataCenterArr[0].toLowerCase() === "eu") {
-      destination.Config.endPoint = "https://rest.fra-01.braze.eu";
+      destination.Config.endPoint = `https://rest.fra-${dataCenterArr[1]}.braze.eu`;
```
Comment (ItsSudip) [inline v0/destinations/braze/transform.js:378]: can't we provide a check for correct number? Let's say someone provided `EU-03`.

Comment (arnab-p) [reply v0/destinations/braze/transform.js:378]: Let's address that from the UI

**Review by arnab-p — APPROVED** [review]

File: v0/destinations/braze/transform.js (hunk 1 above)
Comment (utsabc) [reply v0/destinations/braze/transform.js:378]: We will do the UI change as well but for backward compatibility and future proofing lets add this

**Review by arnab-p — APPROVED** [review]

**Comment by utsabc** [comment]
@ItsSudip can you approve this PR

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**